File: SpringTILCode/src/test/java/com/example/springtilcode/user/controller/UserControllerTest.java
package com.example.springtilcode.user.controller;

import com.example.springtilcode.domain.user.controller.dto.request.EditProfileRequest;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.MediaType;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;

import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@SpringBootTest
@AutoConfigureMockMvc
public class UserControllerTest {
    @Autowired
    private MockMvc mockMvc;

    @Autowired
    private ObjectMapper objectMapper;

    @Test
    public void getUser() throws Exception {
        mockMvc.perform(
                MockMvcRequestBuilders.get("/mypage")
                        .header("Authorization", "Bearer eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0ZXN0QGdtYWlsLmNvbSIsImlhdCI6MTY3NDAxMDExMSwiZXhwIjoxNjc0MDEzNzExfQ.WAcsoOOMPSTZkK1RnuehpZn7oSjzaxs406Z3_nyooO0")
        ).andExpect(status().isOk());
    }

    @Test
    public void editProfile() throws Exception{
        String email = "wheeabc@gmail.com";

        EditProfileRequest request = new EditProfileRequest(email);
        String json = objectMapper.writeValueAsString(request);

        mockMvc.perform(
                MockMvcRequestBuilders.patch("/mypage")
                        .header("Authorization", "Bearer eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0ZXN0QGdtYWlsLmNvbSIsImlhdCI6MTY3NDUzMTE0OCwiZXhwIjoxNjc0NTM0NzQ4fQ.rT3xRGfYNVk0qMf_5WEv_zAoqdop14OUaRmTrKZ2eVU")
                        .content(json)
                        .contentType(MediaType.APPLICATION_JSON)
                        .accept(MediaType.APPLICATION_JSON)
        ).andExpect(status().isOk());
    }
}

File: kotlin/src/test/kotlin/com/example/demo/user/controller/UserControllerTest.kt
package com.example.demo.user.controller

import com.example.demo.domain.user.dto.request.EditProfileRequest
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

@SpringBootTest
@AutoConfigureMockMvc
class UserControllerTest (
    @Autowired
    val mockMvc: MockMvc
){
    @Test
    fun getUser(){
        mockMvc.perform(
            MockMvcRequestBuilders.get("/user")
                .header("Authorization", "bearer eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3aGVlQGdtYWlsLmNvbSIsImV4cCI6MTY3MzY2MTk2OX0.PxM1iNRKBl3fqcum56V7Xiju3PnhNTTmxYbgD6-12sY")
        )
            .andExpect(status().isOk)
    }

    @Test
    fun editProfile(){
        val request = EditProfileRequest("whee1234@gmail.com")
        val json = jacksonObjectMapper().writeValueAsString(request)

        mockMvc.perform(
            MockMvcRequestBuilders.patch("/user")
                .content(json)
                .contentType(MediaType.APPLICATION_JSON)
                .accept(MediaType.APPLICATION_JSON)
                .header("Authorization", "bearer eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3aGVlQGdtYWlsLmNvbSIsImV4cCI6MTY3Mzc5NDI1NH0.dgqn2Y4jtNJvoXBYxeFeDV6Ujsll4cN6GPUR_pgd_5o")
        )
            .andExpect(status().isOk)
    }
}